internal/transformer-client/client.go

```diff
@@ -96,6 +109,67 @@ func NewClient(config *ClientConfig) Client {
 	}
 }
 
+func NewRetryableClient(config *RetryableClientConfig) RetryableClient {
```

Comment:
lot of duplication in this file. 
why can't we use existing logic in NewClient for http client creation & reuse it here? Have a common method to create http client based on ClientConfig and use it in both `NewClient` and `NewRetryableClient`